sources/packages/backend/libs/integrations/src/services/disbursement-value/disbursement-value.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from "@nestjs/common";
+import { MoreThan, Repository } from "typeorm";
+import { DisbursementValue } from "@sims/sims-db";
+import { InjectRepository } from "@nestjs/typeorm";
+
+/**
+ * Service layer for Disbursement value.
+ */
+@Injectable()
+export class DisbursementValueService {
+  constructor(
+    @InjectRepository(DisbursementValue)
+    private readonly disbursementValueRepo: Repository<DisbursementValue>,
+  ) {}
+
+  /**
+   * Checks if there is any partial or full award amount, that
+   * was withheld due to a restriction during an e-cert generation.
+   * @param disbursementScheduleId disbursement schedule id.
+   * @returns true if there is any partial or full award amount, that
+   * was withheld due to a restriction.
+   */
+  async hasAnyFullOrPartialAwardWithheldDueToRestriction(
```

Comment:
The method can be named as `hasAwardWithheldDueToRestriction` and still keeps its meaning, IMO.